src/abstractdataframe/join.jl

```diff
@@ -41,7 +42,27 @@ struct DataFrameJoiner
                                     "Symbol or Pair{Symbol,Symbol}."))
             end
         end
-        new(dfl, dfr, dfl[!, left_on], dfr[!, right_on], left_on, right_on)
+        dfl_on = dfl[!, left_on]
+        dfr_on = dfr[!, right_on]
+
+        if matchmissing === :error
+            for df in (dfl_on, dfr_on), col in eachcol(df)
+                if any(ismissing, col)
+                    throw(ArgumentError("missing values in key columns are not allowed " *
+                                        "when matchmissing == :error"))
+                end
+            end
+        elseif matchmissing !== :equal
+            throw(ArgumentError("matchmissing allows only :error or :equal"))
+        end
+
+        for df in (dfl_on, dfr_on), col in eachcol(df)
+            if any(x -> (x isa Union{Complex, Real}) & isnan(x), col)
```

Comment:
Short-circuiting is needed AFAICT:
```suggestion
            if any(x -> (x isa Union{Complex, Real}) && isnan(x), col)
```

Reply:
indeed